dot/node.go

```diff
@@ -313,47 +353,23 @@ func NewNode(cfg *Config, ks *keystore.GlobalKeystore) (*Node, error) {
 		metrics.PublishMetrics(address)
 	}
 
-	gd, err := stateSrvc.Base.LoadGenesisData()
-	if err != nil {
-		return nil, err
-	}
+	return node, nil
+}
 
+func setupTelemetry(cfg *Config, genesisData *genesis.Data) (*telemetry.Mailer, error) {
 	var telemetryEndpoints []*genesis.TelemetryEndpoint
-	if len(cfg.Global.TelemetryURLs) == 0 {
-		telemetryEndpoints = append(telemetryEndpoints, gd.TelemetryEndpoints...)
+	if len(cfg.Global.TelemetryURLs) == 0 && genesisData != nil {
+		telemetryEndpoints = append(telemetryEndpoints, genesisData.TelemetryEndpoints...)
 	} else {
 		telemetryURLs := cfg.Global.TelemetryURLs
 		for i := range telemetryURLs {
 			telemetryEndpoints = append(telemetryEndpoints, &telemetryURLs[i])
 		}
 	}
 
-	if !cfg.Global.NoTelemetry {
-		telemetryLogger := log.NewFromGlobal(log.AddContext("pkg", "telemetry"))
-
-		err := telemetry.BootstrapMailer(context.Background(), telemetryEndpoints, telemetryLogger)
-		if err != nil {
-			return nil, fmt.Errorf("cannot bootstrap mailer: %w", err)
-		}
-
-		genesisHash := stateSrvc.Block.GenesisHash()
-		connectedMsg := telemetry.NewSystemConnectedTM(
-			cfg.Core.GrandpaAuthority,
-			sysSrvc.ChainName(),
-			&genesisHash,
-			sysSrvc.SystemName(),
-			cfg.Global.Name,
-			networkSrvc.NetworkState().PeerID,
-			strconv.FormatInt(time.Now().UnixNano(), 10),
-			sysSrvc.SystemVersion())
-
-		err = telemetry.SendMessage(connectedMsg)
-		if err != nil {
-			logger.Debugf("problem sending system.connected telemetry message: %s", err)
-		}
-	}
-
-	return node, nil
+	telemetryLogger := log.NewFromGlobal(log.AddContext("pkg", "telemetry"))
+	return telemetry.BootstrapMailer(context.Background(),
```

Comment:
Should you pass a context here instead of `context.Background()`?

Reply:
@qdm12 you mean to create a context outside of that function? I'm using `context.Background()` as there is no available context provided by that function caller, however I can create but the context will be used only by this function (`setupTelemetry`)

Reply:
Interesting. Should I create an issue to review all usage of `context.Background()` and refactor code to pass down contextes? :thinking:

Reply:
Will be a nice issue!

Reply:
Created https://github.com/ChainSafe/gossamer/issues/2145 although it's kind of a recursive refactoring until we reach back to `dot/services.go` really :thinking: